tests/migration/version_switch_upgrade_target.pm

```diff
@@ -43,12 +43,11 @@ sub run {
     }
 
     # Only 11-SP4 need set username, 11-SP4+ don't need set username
-    set_var('DM_NEEDS_USERNAME', '0') if (is_sle('=11-SP4', get_var('HDDVERSION')) && check_var('DM_NEEDS_USERNAME', '1'));
-
-    # Setup DM_NEEDS_USERNAME for SLE15 KDE migration case
-    # In SLES15 KDE has been drop, after migration the default desktop is XDM
-    # KDE has moved to Package Hub, it will stay install with SLES15 if added PackageHub
-    set_var('DM_NEEDS_USERNAME', '1') if (check_var('DESKTOP', 'KDE') && is_sle('15+') && (get_var('ADDONURL', '') !~ /phub/));
+    # Reset DESKTOP after upgrade as desktop change
+    if (is_sle('=11-SP4', get_var('HDDVERSION')) && check_var('DM_NEEDS_USERNAME', '1')) {
+        set_var('DM_NEEDS_USERNAME', '0');
```

Comment:
Previously, `DM_NEEDS_USERNAME` was set to `1` on migration (line 51), but now it's set to `0`. Is that intentional?
Might have to be coupled with the desktop change

Reply:
Only 11sp4 should set DM_NEEDS_USERNAME=1. After upgrade, the system become sles11sp+, it should set DM_NEEDS_USERNAME=0. This is expected.

Reply:
Previously it was set to `1` after upgrade though

Reply:
That code don't work. I checked the case vars.json file, the setting DM_NEEDS_USERNAME=0. So it didn't set to 1 after upgrade.

Reply:
Checked openqa code:
sub check_var {
    my ($var, $val) = @_;
    return 1 if (defined $bmwqemu::vars{$var} && $bmwqemu::vars{$var} eq $val);
    return 0;
}
It will compare check_vars('DESKTOP','KDE'). As it is upper, our setting is DESKTOP=kde,  So those code never be used. It don't impact others.